packages/components/src/popover/index.js

```diff
@@ -42,405 +50,156 @@ const SLOT_NAME = 'Popover';
-		 * are not handled by the resize/scroll window events (i.e. CSS changes
-		 * in the layout that changes the position of the anchor).
-		 *
-		 * For these situations, we refresh the popover every 0.5s
-		 */
-		const intervalHandle = defaultView.setInterval( refresh, 500 );
-
-		let rafId;
-
-		const refreshOnAnimationFrame = () => {
-			defaultView.cancelAnimationFrame( rafId );
-			rafId = defaultView.requestAnimationFrame( refresh );
-		};
-
-		// Sometimes a click trigger a layout change that affects the popover
-		// position. This is an opportunity to immediately refresh rather than
-		// at the interval.
-		defaultView.addEventListener( 'click', refreshOnAnimationFrame );
-		defaultView.addEventListener( 'resize', refresh );
-		defaultView.addEventListener( 'scroll', refresh, true );
-
-		const anchorDocument = getAnchorDocument( anchorRef );
-
-		// If the anchor is within an iframe, the popover position also needs
-		// to refrest when the iframe content is scrolled or resized.
-		if ( anchorDocument && anchorDocument !== ownerDocument ) {
-			anchorDocument.defaultView.addEventListener( 'resize', refresh );
-			anchorDocument.defaultView.addEventListener(
-				'scroll',
-				refresh,
-				true
-			);
+		if ( ! frameElement || ownerDocument === document ) {
+			return undefined;
 		}
 
-		let observer;
-
-		if ( __unstableObserveElement ) {
-			observer = new defaultView.MutationObserver( refresh );
-			observer.observe( __unstableObserveElement, { attributes: true } );
-		}
-
-		return () => {
-			defaultView.clearInterval( intervalHandle );
-			defaultView.removeEventListener( 'resize', refresh );
-			defaultView.removeEventListener( 'scroll', refresh, true );
-			defaultView.removeEventListener( 'click', refreshOnAnimationFrame );
-			defaultView.cancelAnimationFrame( rafId );
-
-			if ( anchorDocument && anchorDocument !== ownerDocument ) {
-				anchorDocument.defaultView?.removeEventListener(
-					'resize',
-					refresh
-				);
-				anchorDocument.defaultView?.removeEventListener(
-					'scroll',
-					refresh,
-					true
-				);
-			}
-
-			if ( observer ) {
-				observer.disconnect();
-			}
+		const iframeRect = frameElement.getBoundingClientRect();
+		return {
+			name: 'iframeOffset',
+			fn( { x, y } ) {
+				return {
+					x: x + iframeRect.left,
+					y: y + iframeRect.top,
+				};
+			},
 		};
-	}, [
-		isExpanded,
-		anchorRect,
-		getAnchorRect,
-		anchorRef,
-		position,
-		contentSize,
-		__unstableStickyBoundaryElement,
-		__unstableObserveElement,
-		__unstableBoundaryParent,
-	] );
+	}, [ anchorRef, anchorRect, getAnchorRect ] );
+
+	const middlewares = [
+		frameOffset,
+		offset ? offsetMiddleware( offset ) : undefined,
+		__unstableForcePosition ? undefined : flip(),
+		__unstableForcePosition
+			? undefined
+			: size( {
+					apply( { width, height } ) {
+						if ( ! refs.floating.current ) return;
+
+						Object.assign( refs.floating.current.firstChild.style, {
+							maxWidth: `${ width }px`,
```

Comment:
I think this is causing the Site Editor's navigation button tooltip to be _cut_.

![image](https://user-images.githubusercontent.com/7753001/170440212-3d1fb628-7d4b-46b3-b7c0-fd540fca0c56.png)

The expected width is supposed to be `115.63px` but the applied maxWidth is `96px` here.

Reply:
Oh, I want to clarify that I don't know if this was an issue already before this PR though!